spec/spec_helper_acceptance_local.rb

```diff
@@ -10,13 +10,22 @@
 # in parallel. For example, what happens if two targets
 # try to modify site.pp at the same time?
 def set_sitepp_content(manifest)
+  site_pp_path = '/opt/puppetlabs/server/data/code-manager/code/environments/production/manifests/site.pp'
+
   content = <<-HERE
   node default {
     #{manifest}
   }
   HERE
 
-  master.run_shell("echo '#{content}' > /etc/puppetlabs/code/environments/production/manifests/site.pp")
+  Tempfile.open('manifest') do |f|
```

Comment:
Why is here?

Reply:
It's easier to create a temp file and upload it than it was to figure out how to properly prefix a command with `run_shell` with `/bin/bash -c''` and all the escaping you have to do.

Reply:
Gotcha. So is this an issue with `run_shell` in general for the containerized PE stuff or _just_ this `site.pp` code? Guess better Q. is is this hiding a more general issue whose fix we should maybe centralize?

Reply:
The problem is in the bolts `docker` transport. `run_shell` gives the command to a transport which is responsible for setting up the session. The winrm transport for instance is a gem that automatically ensures you are running in a PowerShell session on the other end. The docker transport is not so nice to us. It would be nice if they fixed it, but it's not something I'm going to concentrate on right now.

Reply:
So "fix we should maybe centralize" also means (for now) a `run_shell` method in https://github.com/puppetlabs/puppetlabs-servicenow_cmdb_integration/blob/master/spec/support/acceptance/helpers.rb#L27 that is something like

```
if using Docker transport
  // Your fix for run_shell that handles proper escaping. I think this was "create tempfile" => "run script" ?
else
  // Call PuppetLitmus.run_shell
end
```

It'd be nice to have this there to avoid future headaches, e.g. me trying `run_shell`, noticing it not working in CI, and after a few hours tracing the cause back to something you figured out a while back.